### ISSUE TITLE
Add a request() argument to modify the HttpRequest.META dictionary

### ISSUE THREAD

**Issue by seav** [Cadasta/django-skivvy#15]
Currently, there is no way to set additional `HttpRequest.META` dictionary entries in django-skivvy, such as `HTTP_REFERER`. It would be great if these can be set via an additional argument to `ViewTestCase.request()`.

**Comment by oliverroick** [comment]
Can you provide me with an example so I can write a test case as well.

**Comment by seav** [comment]
I guess a trivial example would be if `request.META['HTTP_USER_AGENT']` contains "Googlebot" then the view would inject a context variable that modifies the page to include additional search keywords.

A more practical example would be if `request.META['HTTP_REFERER']` is a Google search engine  results page, the view would inject a context variable that highlights the Google search terms found in the page.

**Comment by oliverroick** [comment]
I released an alpha (`0.1.3a1`) version the implements what you have asked for. Can you please install it on your VM and check if it works as expected? When everything is fine, I'll do a stable release later today, which you can use for your PR.

As always, you can define the settings using a dictionary in different ways:

*As a test class attribute:*

```python
class MyTestCase(ViewTestCase, TestCase):
    request_meta = {'HTTP_REFERER': 'http://example.com'}
```


*As a setup method:*

```python
class MyTestCase(ViewTestCase, TestCase):
    def setup_request_meta(self):
        return {'HTTP_REFERER': 'http://example.com'}
```

*Update individual requests:*

```python
class MyTestCase(ViewTestCase, TestCase):
    request_meta = {'HTTP_REFERER': 'http://example.com'}

    def test_something(self):
        response = self.request(request_meta={'HTTP_REFERER': 'http://example.com/blah'})
```

**Comment by seav** [comment]
The alpha release works as expected, though I only used the individual requests option and not the attribute or method options.

**Comment by oliverroick** [comment]
Implemented in #16 and released as version `0.1.3`
